recirq/qml_lfe/sample_analysis.ipynb

```diff
@@ -0,0 +1,1451 @@
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "LOSS 0.03283559548246558\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      " 92%|███████████████████████████████████▊   | 2751/3000 [13:49<31:55,  7.69s/it]"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "ACC 1.0\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      " 93%|████████████████████████████████████▍  | 2802/3000 [13:55<00:21,  9.22it/s]"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "LOSS 0.027417695932817877\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      " 97%|█████████████████████████████████████▋ | 2900/3000 [14:05<00:08, 11.27it/s]"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "LOSS 0.018936340993477643\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████████████████████████████████▉| 2999/3000 [14:18<00:00,  4.79it/s]"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "LOSS 0.02224759492623889\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|███████████████████████████████████████| 3000/3000 [15:09<00:00,  3.30it/s]"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "ACC 1.0\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "from tqdm import tqdm\n",
+    "import cirq\n",
+    "import numpy as np\n",
+    "from os import listdir\n",
+    "import random\n",
+    "from os.path import isfile, join\n",
+    "from cirq import Simulator\n",
+    "simulator = Simulator()\n",
+    "\n",
+    "all_paulis = dict([(\"I\",0), (\"X\",1), (\"Y\",2), (\"Z\",3)])\n",
+    "\n",
+    "def entire_process_predict_Pauli_observable(qubits, hidden_P, num_rep):\n",
+    "    n = len(qubits) // 2\n",
+    "    output = []\n",
+    "    \n",
+    "    sign_p = -1 if np.random.random() < 0.5 else 1\n",
+    "    for _ in range(num_rep):\n",
+    "        circuit = cirq.Circuit()\n",
+    "\n",
+    "        last_i = 0\n",
+    "        for i, pauli in enumerate(hidden_P):\n",
+    "            if pauli != \"I\":\n",
+    "                last_i = i\n",
+    "            \n",
+    "        for twocopy in [0, 1]:\n",
+    "            parity = sign_p\n",
+    "            for i, pauli in enumerate(hidden_P):\n",
+    "                if pauli != \"I\":\n",
+    "                    if last_i == i:\n",
+    "                        if parity == -1:\n",
+    "                            circuit += cirq.X(qubits[2*i+twocopy])\n",
```

Comment:
Should we make `2*i+twocopy`into a variable.  I think that might help code clarity slightly too.